### PR TITLE
Archive old pages on R2 before deleting old data.

### PR DESCRIPTION
**Pull request by johnwarden**
We currently delete data older than 30 days to keep our database from growing without bound. But the result is we miss out on a lot of traffic, especially from Google, when these turn into broken links. So try archiving our stats page data on R2. Some quick calculations suggest this should be pretty cheap, running into a few dollars a month eventually.

Archiving old stories:
- Archive stories where the submission time is more than 30 days ago
- Before archiving a story, upload JSON file with all stats data to an R2 bucket
- Mark story as archived in DB. Delete all rows from dataset except the latest (so we keep the final story stats such as upvoteRate)
- When a request to stats page is made, if story is marked as archived, read JSON from R2.

R2 Buckets and Keys:
- Separate R2 bucket for dev and prod: news-archive and news-archive-dev
- Also separate access keys.
- Prod access keys are stored in fly.io secrets.
- Dev keys go in .env.local (ask jwarden).

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -93,6 +93,12 @@ func main() {
 func (app app) mainLoop(ctx context.Context) {
 	logger := app.logger
 
+	logger.Info("Arching old stories")
```
Comment (fdietze) [inline main.go:96]:
```suggestion
	logger.Info("Archiving old stories")
```

**Comment by fdietze** [comment]
Very good! Question: Why do we archive the stats data instead of the raw data we delete from the database? Is there a huge difference in size? The raw data would have more use-cases in the future than just displaying the plots. Additionally, the code which derives the plotting data from the raw data would be just the same.

I think my key question is the one about the size of the plot-json vs raw-data csv.

**Comment by johnwarden** [comment]
> Very good! Question: Why do we archive the stats data instead of the raw data we delete from the database? Is there a huge difference in size? The raw data would have more use-cases in the future than just displaying the plots. Additionally, the code which derives the plotting data from the raw data would be just the same.
> 
> I think my key question is the one about the size of the plot-json vs raw-data csv.

I started by archiving the HTML and then doing a 302 redirect to the archived page, because that seemed like the simplest solution and let us leverage the cloudflare CDN. But then I realized it would be better to archive just the data. But now that you mention it a CSV of the actual rows deleted seems like would be even better.

**Comment by johnwarden** [comment]
I am going to merge this as it is because I don't want to spend too much more time on it, and to properly archive old data as CSV would take a little more refactoring (e.g. functions like rankDatapoints would need to get the data from a CSV instead of a sqlite query).

**Pull request merged**